src/boot/efi/console.c

```diff
@@ -193,6 +193,21 @@ EFI_STATUS console_set_mode(UINT32 mode) {
                         return change_mode(CONSOLE_MODE_MIN);
                 return EFI_SUCCESS;
 
+        case CONSOLE_MODE_NEXT:
+                if (ST->ConOut->Mode->MaxMode <= CONSOLE_MODE_MIN)
+                        return EFI_UNSUPPORTED;
+
+                mode = ST->ConOut->Mode->Mode;
+                do {
+                        mode = (mode + 1) % ST->ConOut->Mode->MaxMode;
```

Comment:
if mode is UINT32_MAX this will overflow. mostly theoretical overflow I admit, but I'd prefer if we'd catch it anyway.